src/testing/time_refinement_study.cpp

```diff
@@ -0,0 +1,130 @@
+{}
+
+
+
+template <int dim, int nstate>
+Parameters::AllParameters TimeRefinementStudy<dim,nstate>::reinit_params_and_refine_timestep(int refinement) const
+{
+     PHiLiP::Parameters::AllParameters parameters = *(this->all_parameters);
+     
+     parameters.ode_solver_param.initial_time_step *= pow(refine_ratio,refinement);
+     
+     return parameters;
+}
+
+template <int dim, int nstate>
+double TimeRefinementStudy<dim,nstate>::calculate_L2_error_at_final_time_wrt_function(std::shared_ptr<DGBase<dim,double>> dg, const Parameters::AllParameters parameters) const
+{
+
+    //generate exact solution at final time
+    std::shared_ptr<ExactSolutionFunction<dim,nstate,double>> exact_solution_function;
+    exact_solution_function = ExactSolutionFactory<dim,nstate,double>::create_ExactSolutionFunction(parameters.flow_solver_param);
+    int poly_degree = parameters.grid_refinement_study_param.poly_degree;
+    dealii::Vector<double> difference_per_cell(dg->solution.size());
+    
+    dealii::VectorTools::integrate_difference(dg->dof_handler,    
+                                              dg->solution,
+                                              *exact_solution_function,
+                                              difference_per_cell,
+                                              dealii::QGauss<dim>(poly_degree+10), //overintegrating by 10
+                                              dealii::VectorTools::L2_norm);
+
+    double L2_error = dealii::VectorTools::compute_global_error(*dg->triangulation,
+                                                              difference_per_cell,
+                                                              dealii::VectorTools::L2_norm);
+    return L2_error;    
+
+}
+
+
+template <int dim, int nstate>
+int TimeRefinementStudy<dim, nstate>::run_test() const
+{
+
+    double final_time = this->all_parameters->flow_solver_param.final_time;
+    double initial_time_step = this->all_parameters->ode_solver_param.initial_time_step;
+    int n_steps = round(final_time/initial_time_step);
+    if (n_steps * initial_time_step != final_time){
+        pcout << "Error: final_time is not evenly divisible by initial_time_step!" << std::endl
+              << "Remainder is " << fmod(final_time, initial_time_step)
+              << ". Modify parameters to run this test." << std::endl;
+        std::abort();
+    }
+
+    int testfail = 0;
+    double expected_order =(double) this->all_parameters->ode_solver_param.runge_kutta_order;
+    double order_tolerance = 0.1;
+
+    dealii::ConvergenceTable convergence_table;
+    double L2_error_old = 0;
+    double L2_error_conv_rate=0;
+
+    for (int refinement = 0; refinement < n_time_calculations; ++refinement){
+        
+        pcout << "\n\n---------------------------------------------\n Refinement number " << refinement <<
+            " of " << n_time_calculations - 1 << std::endl << "---------------------------------------------" << std::endl;
```

Comment:
clean this up into separate `pcout` statements

Reply:
Okay, will modify.